src/Clip.cpp

```diff
@@ -150,6 +150,10 @@ Clip::Clip(ReaderBase* new_reader) : resampler(NULL), reader(new_reader), alloca
 
 	// Update duration
 	End(reader->info.duration);
+
+	if (new_reader) {
+		new_reader->SetClip(this);
+	}
```

Comment:
Regarding this, for consistency with the rest of the code it's probably better to go with,
```suggestion
	reader->SetClip(this);
```
since the initializer list has already set `reader` to `new_reader`, and the previous line already assumes a valid `reader`.

(Or you could wrap the whole thing, including the preceding `End(reader->info.duration);`, in `if (reader) { }` for safety. Which wouldn't be a bad idea at all. But Github doesn't allow suggested changes that touch lines outside of the commit, stupidly.)

Also, if you'd be willing to add a `reader->SetClip(this);` to this part of the `Clip::Clip(std::string path)` constructor, too, that'd be appreciated. Makes sense for them all to be consistent.
https://github.com/OpenShot/libopenshot/blob/4e6c1819b0af7e33ac27d22fa02a09ad268f25b1/src/Clip.cpp#L209-L214

Reply:
I have made this change as requested.